internal/utils/utils.go

```diff
@@ -64,7 +63,8 @@ func ConvertFixedDataIntoByteArray(data interface{}) []byte {
 	buff := new(bytes.Buffer)
 	err := binary.Write(buff, binary.BigEndian, data)
 	if err != nil {
-		log.Panic(err)
+		GetLogger().Crit("Failed to convert fixed data into byte array", "err", err)
+		panic(err)
```

Comment:
don't think we need to panic here.  @mikedoan

Reply:
Not sure about the context here, but what's the damage if it fails to write?